test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,679 @@
+            input = make_video(size=self.INPUT_SIZE, dtype=dtype or torch.uint8, device=device)
+
+        return input
+
+    def _check_size(self, input, output, *, size, max_size):
+        if isinstance(size, int) or len(size) == 1:
+            if not isinstance(size, int):
+                size = size[0]
+
+            old_height, old_width = F.get_spatial_size(input)
+            ratio = old_width / old_height
+            if ratio > 1:
+                new_height = size
+                new_width = int(ratio * new_height)
+            else:
+                new_width = size
+                new_height = int(new_width / ratio)
+
+            if max_size is not None and max(new_height, new_width) > max_size:
+                # Need to recompute the aspect ratio, since it might have changed due to rounding
+                ratio = new_width / new_height
+                if ratio > 1:
+                    new_width = max_size
+                    new_height = int(new_width / ratio)
+                else:
+                    new_height = max_size
+                    new_width = int(new_height * ratio)
+
+        else:
+            new_height, new_width = size
+
+        assert F.get_spatial_size(output) == [new_height, new_width]
+
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    @pytest.mark.parametrize("interpolation", INTERPOLATION_MODES)
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_image_tensor(self, size, interpolation, use_max_size, antialias, dtype, device):
+        if not (max_size_kwarg := self._make_max_size_kwarg(use_max_size=use_max_size, size=size)):
+            return
+
+        check_kernel(
+            F.resize_image_tensor,
+            self._make_input(datapoints.Image, dtype=dtype, device=device),
+            size=size,
+            interpolation=interpolation,
+            **max_size_kwarg,
+            antialias=antialias,
+            # The `InterpolationMode.BICUBIC` implementation on CUDA does not match PILs implementation well. Thus,
+            # instead of testing with an enormous tolerance, we disable the check all together.
+            check_cuda_vs_cpu=False
+            if interpolation is transforms.InterpolationMode.BICUBIC
+            else dict(rtol=0, atol=1 / 255 if dtype.is_floating_point else 1),
+            check_scripted_vs_eager=not isinstance(size, int),
+        )
+
+    @pytest.mark.parametrize("size", OUTPUT_SIZES)
+    @pytest.mark.parametrize("format", list(datapoints.BoundingBoxFormat))
+    @pytest.mark.parametrize("use_max_size", [True, False])
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.int64])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_bounding_box(self, size, format, use_max_size, dtype, device):
```

Comment:
format is never used, it's hardcoded in `_make_input()`

Reply:
Good catch!